Add com.github.Cogitri.gxi

Cogitri opened the pull request:
Adds a flatpak manifest for https://github.com/Cogitri/gxi

nedrichards commented on com.github.Cogitri.gxi.yml:
```diff
@@ -0,0 +1,28 @@
+  - org.freedesktop.Sdk.Extension.rust-stable
+command: gxi
+finish-args:
+  - --socket=x11
```
you need to --share=ipc if you use x11 for full performance

nedrichards commented on com.github.Cogitri.gxi.yml:
```diff
@@ -0,0 +1,28 @@
+build-options:
+  append-path: /usr/lib/sdk/rust-stable/bin
+  build-args:
+  - --share=network
```
access to the network during the build is banned on flathub. you need to specify all dependencies in the manifest.

nedrichards replied:
I believe that's what the other rust apps do (https://github.com/search?q=org%3Aflathub+rust&type=Code can show you those)

Cogitri replied:
Yup, thanks, had looked at Podcasts before. Currently building...

Cogitri commented:
Updated, please try again.

Cogitri commented on com.github.Cogitri.gxi.yml:
```diff
@@ -0,0 +1,27 @@
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --filesystem=home
```
Actually, I don't think this is necessary, as I use GtkFileChooser for opening files. Will figure out why it doesn't show any files without this.

Cogitri commented:
I'm not quite sure if I read https://github.com/flathub/flathub/wiki/App-Requirements#Appstream right. Is CC0-1.0 required for the Appstream file, or is MIT alright too? I wouldn't mind relicensing it.

nedrichards commented:
Which Nvidia endpoints does this need? If it uses [these bits](https://github.com/flatpak/flatpak/commit/e5218a56239bc7ea22873e245da99e1470ea7fb7) then it needs a minimum of Flatpak 1.1.1 (which you can specify in the manifest)

Cogitri commented:
> Which Nvidia endpoints does this need? If it uses these bits then it needs a minimum of Flatpak 1.1.1 (which you can specify in the manifest)

I don't think it needs anything Nvidia specific

barthalion commented:
It doesn't not start for me – just hangs without any error. strace output: https://paste.xinu.at/zra/

Cogitri commented:
Ah. Yes, it seems like the locking in the rpc thread goes wrong sometimes, am working on it. Thanks for the strace output! :)

Cogitri commented:
Hey, could you try again with https://github.com/Cogitri/gxi/pull/31.patch?

Cogitri commented:
Alright, will release a new version soon.

Cogitri commented:
Updated to 0.5.1,

nedrichards commented:
bot, build com.github.Cogitri.gxi
testbot, build com.github.Cogitri.gxi

Cogitri commented:
Hum, could it be possible that Flathub's Rust extensions doesn't quite work anymore? Tested with both gxi v0.4.6 (which has built before) and v0.5.1, which both don't build anymore (due to rustc SIGSEVing) in the flatpak environment but work fine everywhere else (CI, locally, in Void Linux' packaging)

nedrichards commented:
bot, build com.github.Cogitri.gxi

Cogitri commented:
Hey, does flathub use Ubuntu's system LLVM or the Rust vendored LLLVM? Seems like Void Linux has the same problems with system LLVM (7)

danigm commented:
I've tried to build in my machine and I'm having the same error that we can see in the buildbot. There's a problem building xi-core-lib.

If I try to build with `cargo build` inside the flatpak-builder sandbox, with my local rust, instead of the rust version from the extension, it's working, so maybe the problem is in the `org.freedesktop.Sdk.Extension.rust-stable`.

So maybe we should ask to the freedesktop rust-extension maintainers to know what's happening here.

This is the output with the --verbose flag for cargo:

```
Caused by:
  process didn't exit successfully: `rustc --edition=2018 --crate-name gxi src/main.rs --color always --crate-type bin --emit=dep-info,link -C opt-level=3 -C lto -C codegen-units=1 -C metadata=8a7571c4f424c27d -C extra-filename=-8a7571c4f424c27d --out-dir /run/build/gxi-1/target/release/deps -L dependency=/run/build/gxi-1/target/release/deps --extern cairo=/run/build/gxi-1/target/release/deps/libcairo-8e389cb4c3dcc4d2.rlib --extern cairo_sys=/run/build/gxi-1/target/release/deps/libcairo_sys-f18b2ec8a4d846d0.rlib --extern crossbeam_deque=/run/build/gxi-1/target/release/deps/libcrossbeam_deque-56ecbcb380ef8156.rlib --extern dirs=/run/build/gxi-1/target/release/deps/libdirs-d684b6f8e4b5b9a5.rlib --extern env_logger=/run/build/gxi-1/target/release/deps/libenv_logger-87d0915e357f2e6a.rlib --extern failure=/run/build/gxi-1/target/release/deps/libfailure-3792409224252ffe.rlib --extern gdk=/run/build/gxi-1/target/release/deps/libgdk-447b99d4c74621bb.rlib --extern gettextrs=/run/build/gxi-1/target/release/deps/libgettextrs-391424368c9e76dd.rlib --extern gio=/run/build/gxi-1/target/release/deps/libgio-83ea88834f24a3a9.rlib --extern glib=/run/build/gxi-1/target/release/deps/libglib-20ae9af6467e7327.rlib --extern glib_sys=/run/build/gxi-1/target/release/deps/libglib_sys-70f10b2d39734928.rlib --extern gobject_sys=/run/build/gxi-1/target/release/deps/libgobject_sys-607988e171dc7e43.rlib --extern gtk=/run/build/gxi-1/target/release/deps/libgtk-41f13c7713330346.rlib --extern gtk_sys=/run/build/gxi-1/target/release/deps/libgtk_sys-a5c3320a9405afa6.rlib --extern lazy_static=/run/build/gxi-1/target/release/deps/liblazy_static-0d9dd51ff243516d.rlib --extern libc=/run/build/gxi-1/target/release/deps/liblibc-f3ec97d912e0bc8a.rlib --extern log=/run/build/gxi-1/target/release/deps/liblog-20d2c57ef4fe3080.rlib --extern pango=/run/build/gxi-1/target/release/deps/libpango-4b88c7048ab9ae38.rlib --extern pango_sys=/run/build/gxi-1/target/release/deps/libpango_sys-1d95e95b967a9506.rlib --extern pangocairo=/run/build/gxi-1/target/release/deps/libpangocairo-35c6a4b4b9dd6471.rlib --extern serde=/run/build/gxi-1/target/release/deps/libserde-8a0a6bd1f1fa9b99.rlib --extern serde_derive=/run/build/gxi-1/target/release/deps/libserde_derive-a34b28f130c568e7.so --extern serde_json=/run/build/gxi-1/target/release/deps/libserde_json-4a8c47554de3eab4.rlib --extern tempfile=/run/build/gxi-1/target/release/deps/libtempfile-7671ff205ddf44e2.rlib --extern toml=/run/build/gxi-1/target/release/deps/libtoml-253a2e150f5aaefb.rlib --extern xi_core_lib=/run/build/gxi-1/target/release/deps/libxi_core_lib-c423b405d3a38c42.rlib --extern xi_rpc=/run/build/gxi-1/target/release/deps/libxi_rpc-2e1ec9e87871dfd3.rlib -L native=/run/build/gxi-1/target/release/build/backtrace-sys-902af2afb188f2c6/out -L native=/run/build/gxi-1/target/release/build/miniz-sys-bffd334e139de696/out` (signal: 11, SIGSEGV: invalid memory reference)
```

alatiera commented:
This is a weird error indeed. If somebody could run with `gdb` and find out where the `SIGSEGV` is coming from it would be really helpful.

FWIW the Rust extension uses a private copy of `LLVM 7` currently. You can also file an issue about it [here](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/issues)

alatiera commented:
Looks like the crash above is caused related to this [issue](https://github.com/rust-lang/rust/issues/57762). Try disabling LTO for your build and see if it fixes it.

Cogitri commented:
Thanks for testing, that does seem to be it! I'll wait for the patch in flatpak's build then :)

Cogitri commented:
Alright, updated to 0.5.3 and added a patch to build it without LTO, which builds fine now. I do get the following error now though:

```
Error: module gxi: Failed to execute child process “eu-strip” (No such file or directory)
```

I guess elfutils should be in the Sdk (and has been previously)? Not quite sure why that pops up.

nedrichards commented:
bot, build com.github.Cogitri.gxi

Cogitri commented:
What size should the SVG be? 512x512?

Cogitri commented:
Updated to 0.5.4

nedrichards commented:
bot, build com.github.Cogitri.gxi

Cogitri commented:
Oh, the rust sdk bug hasn't been fixed? Will reapply the patch for building without lto then

nedrichards commented:
bot, build com.github.Cogitri.gxi

nedrichards commented:
Repository has been created: https://github.com/flathub/com.github.Cogitri.gxi

Cogitri commented:
Thanks! Since Rust still SIGSEVs sometimes, is it possible to restart builds?

nedrichards commented:
@Cogitri sure, go to https://flathub.org/builds/ and sign in with your github account in the top right and then you can restart builds, kick them off - install test builds or promote them to release sooner than 24 hours.

Cogitri commented:
Oh, great, thanks for the info! :)
